yaspin/core.py

```diff
@@ -319,9 +319,12 @@ def write(self, text):
             sys.stdout.write("\r")
             self._clear_line()
 
-            _text = to_unicode(text)
-            if PY2:
-                _text = _text.encode(ENCODING)
+            if isinstance(text, (builtin_str, str, bytes)):
+                _text = to_unicode(text)
+                if PY2:
+                    _text = _text.encode(ENCODING)
+            else:
+                _text = builtin_str(text)
```

Comment:
I'm planning to drop Py2 support in `yaspin v2.0.0`, thus `yaspin/compat.py` layer will disappear and `builtin_str` will disappear as well. Let's make our intention more *yaspin_v2-ready* and clear:
```suggestion
                _text = str(text)
```

Reply:
Using `str` here breaks python2, as `compat.py` overrides `str` with `unicode`.

If yaspin v2 moves in direction of dropping python2 support then we could do a rewrite of `write()` without all that compat cruft. So I see these options:

1. merge it this way (with PY2 compat garbo)
2. I make this a two-commit PR where I remove the compat layer, then add a much cleaner implementation of my write-with-non-str on top of that
3. extra PR; you or I drop PY2 support as an extra commit
4. further discussion / options I did not see :)

What do you think we shall do?

Reply:
Let's proceed making this PR both Py3 and Py2 compatible, ie. keeping `_text = builtin_str(text)` (will include it in yaspin v1.4.0). Dropping Py2 (yaspin v2) will be a separate PR.